### PR TITLE
Rework Antenna Repair

### PR DESCRIPTION
**Pull request by kareefardi**
\- Remove BM64 test design
\~ Update a couple of CI designs enabling RUN_HEURISTIC_DIODE_INSERTION and DIODE_ON_PORTS
\~ Move OpenROAD DPL logic and API calls to scripts/openroad/common/dpl.tcl and update scripts/openroad/dpl.tcl accordingly
\~ Move OpenROAD GRT logic and API calls to scripts/openroad/common/grt.tcl and update scripts/openroad/grt.tcl accordingly
\~ Use `repair_antennas.tcl` instead of running it inside GRT script
\+ Add scripts/openroad/repair_antennas.tcl with the following flow:
	* GRT - needed to run OR's repair antennas
	* OR's repair antennas (with cell padding for diodes)
	* DPL to legalize
	* GRT to create new guides for legalized cells

[ci ets]

### CODE REVIEW

File: scripts/tcl_commands/routing.tcl [hunk 1]
```diff
@@ -408,6 +407,7 @@ proc run_routing {args} {
         }
         if { $::env(RUN_HEURISTIC_DIODE_INSERTION) } {
             heuristic_diode_insertion
+            detailed_placement_or
```
Comment (donn) [inline scripts/tcl_commands/routing.tcl:410]: Farid told me to leave a comment here idk why

Comment (kareefardi) [reply scripts/tcl_commands/routing.tcl:410]: I thought we might need detailed placement afterwards but turns out that it is already implemented.

**Comment by kareefardi** [comment]
Checking salsa20

**Comment by kareefardi** [comment]
Only aes_core is failing now. An issue is open on github (https://github.com/The-OpenROAD-Project/OpenROAD/issues/3494). 

> Also salsa20 failed due to lvs error (randomly?). I couldn't replicate the error. The problem didn't occur again after rerunning the CI.

**Review by donn — DISMISSED** [review]

**Review by donn — APPROVED** [review]

**Pull request merged**